nrcatalogtools/waveform.py

```diff
@@ -11,10 +11,15 @@
 from nrcatalogtools import utils
 from nrcatalogtools.lvc import (check_interp_req,
                                 get_nr_to_lal_rotation_angles, get_ref_vals)
+from nrcatalogtools.utils import interpolate_in_amp_phase
 from sxs import WaveformModes as sxs_WaveformModes
+from sxs.time_series import TimeSeries as SXSTimeSeries
 from sxs.waveforms.nrar import (h, translate_data_type_to_spin_weight,
                                 translate_data_type_to_sxs_string)
 
+# Add method to SXS TimeSeries class
+SXSTimeSeries.interpolate_in_amp_phase = interpolate_in_amp_phase
```

Comment:
Interesting choice here. Thinking out loud: the `sxs.TimeSeries` class is not aware of what kind of time series it is holding. The concept of amplitude and phase are not available there, as such. Wouldn't this function would be better placed as a member of `WaveformModes` class below, instead of as a standalone method in `utils.py`? I can't really think of another use of it than to interpolate waveform modes / polarizations.

Reply:
Thanks for the suggestion. I was thinking the same.

Reply:
I agree in principle, and then it should really be inside the `sxs` package. Mike would be happy for you to contribute to it directly.